Code/Framework/AzCore/AzCore/Component/EntityId.h

```diff
@@ -19,6 +21,16 @@
 
 namespace AZ
 {
+    class EntityAllocator final
+        : public AZ::ChildAllocatorSchema<AZ::SystemAllocator>
+    {
+        using Base = AZ::ChildAllocatorSchema<AZ::SystemAllocator>;
+    public:
+        AZ_RTTI(EntityAllocator, "{C3FA54B6-DAFC-44A8-98C2-7EB0ACF92BE8}", Base);
+    };
+
+    using EntityStdAllocator = AZ::AZStdAlloc<EntityAllocator>;
```

Comment:
is `EntityStdAllocator` used anywhere?
I wonder if it'd help to macro this so you could call it like:
`AZ_CHILD_ALLOCATOR( <allocator name>, <UUID>, <base allocator> )`

Reply:
EntityStdAllocator is for use with AZStd containers such as an AZStd::vector or AZStd::unordered_map.

Yes we can also add an AZ_CHILD_ALLOCATOR macro to simplify the boilerplate